packages/nextjs-kit/src/types/index.ts

```diff
@@ -0,0 +1,15 @@
+export interface HeaderData {
```

Comment:
Still not wild about this interface name. Let's match it up with the name of the prop which in this case is `navItems`, so we can call this `NavItems`.

Also, we can compose our types a bit more so that instead of `FooterProps`, we have `LinkProps`, and then `FooterMenuItem` can extend `LinkProps` and you can use `LinkProps[]` as the type for the `HeaderProps.navItems`